src/main/java/org/symphonyoss/symphony/messageml/elements/DatePicker.java

```diff
@@ -113,6 +112,12 @@ public void validate() throws InvalidInputException {
 
     if (getAttribute(FORMAT_ATTR) != null) {
       assertAttributeMaxLength(FORMAT_ATTR, DEFAULT_MAX_LENGTH);
+      String format = getAttribute(FORMAT_ATTR);
+      if(!format.matches(dateFormatAllowed)){
+        throw new InvalidInputException(
+            String.format("Attribute \"%s\" contains an unsupported date format, only 'M', 'd' and 'y' are supported with a space or '.','-','/' as separator", FORMAT_ATTR)
+        );
+      }
       try {
         DateTimeFormatter.ofPattern(getAttribute(FORMAT_ATTR));
```

Comment:
as we try to parse the date here, what if we used https://docs.oracle.com/javase/8/docs/api/java/time/LocalDate.html parse method directly?

Reply:
besides that one is JVM locale dependant perhaps we should use it with a forced locale?

Reply:
@symphony-youri  the problem is that we don't parse a date, but we parse a date format. But at the same time, we want to exclude hours and other things that months, days, years.
I have not found any way to do it, as the most common Java utils to parse a date format (DateTimeFormatter, SimpleDateFormat) don't have the option to enforce only dates. I took a look also to other frameworks (apache, joda) but I don't think they can help. For this reason, the easiest way to implements this check is to verify that in the string they are only months, days, years placeholders, then to try to parse it.
For sure it is ugly. Any better idea is really appreciated...

Reply:
Correct I mixed things between parsing the date and parsing the format!
No better idea on my side too then :)